### PR TITLE
Added Cases to handle concurrency during sshpolling

### PR DESCRIPTION
**Pull request by rrpod**

### CODE REVIEW

File: test/unit/services/awsAdapter_spec.js [hunk 1]
```diff
@@ -564,6 +565,157 @@ describe("The AwsAdapter class ", function () {
...
+				});
+			});
+
+			after(function () {
+				SshPollingStub.restore();
+				getPublicIPAddressStub.restore();
+				getInstanceStub.restore();
+			});
+
+			it("fails to update as it has wrong revision number", function () {
```
Comment (a-type) [inline test/unit/services/awsAdapter_spec.js]: I'm not sure I understand this test case. If the `getInstance` indicates that the instance has not been updated, why does the operation fail?

Comment (rrpod) [inline test/unit/services/awsAdapter_spec.js]: This is a test case where a getInstance is performed, an update instance is done (by a PUT request) and then an updateInstance is done based on the details received from the getInstance.

Comment (a-type) [inline test/unit/services/awsAdapter_spec.js]: Perhaps that could be clarified better in the description. Just based on the description alone, it looks like it should succeed.

Comment (rrpod) [inline test/unit/services/awsAdapter_spec.js]: I wanted to make it more descriptive. However multi-line descriptions aren't allowed with the current settings of jslint! Shall I write it in 2 describe blocks?

File: test/unit/services/awsAdapter_spec.js [hunk 2]
```diff
@@ -564,6 +565,157 @@ describe("The AwsAdapter class ", function () {
...
+					SshPollingStub = Sinon.stub(sshAdapter,"SshPolling", function () {
+						return Bluebird.reject("Simulated Failure.");
+					});
+
+					getPublicIPAddressStub = Sinon.stub(awsAdapter,"getPublicIPAddress", function () {
+						return Bluebird.resolve(VALID_IP_ADDRESS);
+					});
+
+					getInstanceStub = Sinon.stub(instances, "getInstance", function () {
+						return instances.updateInstance(instance)
```
Comment (a-type) [inline test/unit/services/awsAdapter_spec.js]: Would be nice to have some inline documentation here to explain what is happening.

File: test/unit/services/awsAdapter_spec.js (hunk 1 above)
Comment (a-type) [inline test/unit/services/awsAdapter_spec.js]: You could break out `"faces an error while polling"` into its own block and make the others sub-descriptions of that. I wouldn't focus so much on `getInstance` in the description, and focus more on the fact that an out-of-band update has occurred.

**Comment by a-type** [comment]
Thanks!

**Pull request merged**